src/TestLogger/Extensions/XunitTestAdapter.cs

```diff
@@ -40,13 +40,13 @@ public class XunitTestAdapter : ITestAdapter
 
             foreach (var result in results)
             {
-                if (skippedTestNamesWithReason.TryGetValue(result.Result.TestCase.DisplayName, out var skipReason))
+                if (skippedTestNamesWithReason.TryGetValue(result.DisplayName, out var skipReason))
```

Comment:
Adapter specific implementation - we should validate if Xunit is copying Displayname to `TestResult` and using it for skip reason instead of `TestCase.Displayname`.

In case of `Theory` test case, a single test case can have multiple `TestResult` and accordingly the `DisplayName` may change too.

Reply:
We now have `TestResultDisplayName` and `TestCaseDisplayName` on the result info. So the string being used here won't have changed.

I made those properties internal so the downstream loggers would be forced to use `Method`. Currently Xunit and Nunit loggers are using  `new XAttribute("name", result.TestCaseDisplayName),` So, I guess those loggers weren't getting the benefit of the ITestAdapters. Not sure how much of an impact it might make on results we output.